scripts/run_acceptance_tests_test.py

```diff
@@ -0,0 +1,413 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/run_acceptance_tests.py."""
+
+from __future__ import annotations
+
+import contextlib
+import subprocess
+import sys
+import time
+
+from core.constants import constants
+from core.tests import test_utils
+from scripts import build
+from scripts import common
+from scripts import run_acceptance_tests
+from scripts import scripts_test_utils
+from scripts import servers
+
+from typing import ContextManager, Optional, Tuple
+
+
+def mock_managed_long_lived_process(
+    *unused_args: str, **unused_kwargs: str
+) -> ContextManager[scripts_test_utils.PopenStub]:
+    """Mock method for replacing the managed_process() functions to simulate a
+    long-lived process. This process stays alive for 10 poll() calls, and
+    then terminates thereafter.
+
+    Returns:
+        Context manager. A context manager that always yields a mock
+        process.
+    """
+    stub = scripts_test_utils.PopenStub(alive=True)
+
+    def mock_poll(stub: scripts_test_utils.PopenStub) -> Optional[int]:
+        stub.poll_count += 1
+        if stub.poll_count >= 10:
+            stub.alive = False
+        return None if stub.alive else stub._return_code    # pylint: disable=protected-access
+
+    # Here we use MyPy ignore because we are assigning a None value
+    # where instance of 'PlatformParameter' is expected, and this is
+    # done to Replace the stored instance with None in order to
+    # trigger the unexpected exception during update.
+    stub.poll = lambda: mock_poll(stub)    # type: ignore[assignment]
+
+    return contextlib.nullcontext(enter_result=stub)
+
+
+def mock_managed_process(
+    *unused_args: str, **unused_kwargs: str
+) -> ContextManager[scripts_test_utils.PopenStub]:
+    """Mock method for replacing the managed_process() functions.
+
+    Returns:
+        Context manager. A context manager that always yields a mock
+        process.
+    """
+    return contextlib.nullcontext(
+        enter_result=scripts_test_utils.PopenStub(alive=False))
+
+
+class RunAcceptanceTestsTests(test_utils.GenericTestBase):
+    """Test the run_acceptance_tests methods."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.exit_stack = contextlib.ExitStack()
+
+        def mock_constants() -> None:
+            print('mock_set_constants_to_default')
+        self.swap_mock_set_constants_to_default = self.swap(
+            common, 'set_constants_to_default', mock_constants)
+
+    def tearDown(self) -> None:
+        try:
+            self.exit_stack.close()
+        finally:
+            super().tearDown()
+
+    def test_wait_for_port_to_be_in_use_when_port_successfully_opened(
```

Comment:
This tests a method in common.py. Why is it in this file?

Ditto below.

Reply:
removed some tests that were not required in this test file